internal/suggestions/suggestions.go

```diff
@@ -270,23 +278,92 @@ func (s *Suggestions) getSuggestionAndRevalidate(validationErr error, previousSu
 	}
 }
 
-func (s *Suggestions) revalidate() ([]error, error) {
-	vErrs, vWarns, vInfo, err := validation.Validate(s.File, s.FilePath, true)
+func (s *Suggestions) revalidate(printSummary bool) ([]errorAndCommentLineNumber, error) {
+	errs, err := validate(s.File, s.FilePath, s.Config.Level, printSummary)
+	if err != nil {
+		return nil, err
+	}
+
+	var errsWithLineNums []errorAndCommentLineNumber
+	if s.FileType == "yaml" {
+		errsWithLineNums, err = s.updateErrsWithYamlLineNums(errs)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		for _, err := range errs {
+			vErr := errors.GetValidationErr(err)
+			if vErr != nil {
+				errsWithLineNums = append(errsWithLineNums, errorAndCommentLineNumber{
+					error:      err,
+					lineNumber: vErr.LineNumber,
+				})
+			}
+		}
+	}
+
+	return errsWithLineNums, nil
+}
+
+func validate(schema []byte, schemaPath string, level errors.Severity, printSummary bool) ([]error, error) {
+	vErrs, vWarns, vInfo, err := validation.Validate(schema, schemaPath, true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate YAML: %v", err)
+	}
+
+	if printSummary {
+		printValidationSummary(vErrs, vWarns, vInfo)
+	}
+
+	toSuggestFor := vErrs
+	switch level {
+	case errors.SeverityWarn:
+		toSuggestFor = append(toSuggestFor, vWarns...)
+	case errors.SeverityHint:
+		toSuggestFor = append(append(toSuggestFor, vWarns...), vInfo...)
+	}
+
+	return toSuggestFor, nil
+}
+
+func (s *Suggestions) updateErrsWithYamlLineNums(errs []error) ([]errorAndCommentLineNumber, error) {
+	yamlFile, err := convertJsonToYaml(s.File)
 	if err != nil {
 		return nil, err
 	}
 
-	errs := vErrs
-	switch s.Config.Level {
-	case "warn":
-		errs = append(errs, vWarns...)
-		break
-	case "info":
-		errs = append(append(errs, vWarns...), vInfo...)
-		break
+	yamlErrs, err := validate(yamlFile, s.FilePath, s.Config.Level, false)
+	if err != nil {
+		return nil, err
 	}
 
-	return errs, nil
+	var newErrs []errorAndCommentLineNumber
+	for _, jsonErr := range errs {
+		vjErr := errors.GetValidationErr(jsonErr)
+		if vjErr == nil {
+			continue
+		}
+
+		for _, yamlErr := range yamlErrs {
+			vyErr := errors.GetValidationErr(yamlErr)
+			if vyErr == nil {
+				continue
+			}
+
+			if validationErrsEqualExceptLineNumber(*vyErr, *vjErr) {
+				newErrs = append(newErrs, errorAndCommentLineNumber{
```

Comment:
Took me a second to figure out what we were trying to do here. I imagine looking back on this when we don't have the context on what the purpose of this PR is we will be confused, some comments might help

Reply:
sounds good, will add them